apps/dashboard/src/components/primitives/url-input.tsx

```diff
@@ -14,23 +13,35 @@ type URLInputProps = Omit<InputProps, 'value' | 'onChange' | 'size'> & {
   options: string[];
   value: URLValue;
   onChange: (value: URLValue) => void;
+  asEditor?: boolean;
 } & Pick<InputFieldProps, 'size'>;
 
 export const URLInput = forwardRef<HTMLInputElement, URLInputProps>((props, ref) => {
-  const { options, value, onChange, size, ...rest } = props;
+  const { options, value, onChange, size = 'default', asEditor = false, placeholder, ...rest } = props;
 
   return (
     <div className="flex items-center justify-between space-x-2">
       <div className="relative flex-grow">
         <InputField className="pr-0" size={size}>
-          <Input
-            ref={ref}
-            type="text"
-            className="min-w-[20ch]"
-            value={value.url}
-            onChange={(e) => onChange({ ...value, url: e.target.value })}
-            {...rest}
-          />
+          {asEditor ? (
+            <Editor
+              size={size}
+              placeholder={placeholder}
+              value={value.url}
+              onChange={(val) => onChange({ ...value, url: val })}
+              height={size === 'md' ? '38px' : '30px'}
```

Comment:
I'm not sure how to do this better without hardcoding these values, basically, our inputs heights are `md 40px` and `default 32px`. These values exclude border. 
And the height should be set if we want to have "single line" input, otherwise the editor will be growing with more content.

Reply:
Maybe I'll move to constants or something.

Reply:
Are we using code mirror in online inputs to reuse the autocomplete functionality for variables? It's a smart approach to keep it dry, although it feels like an overkill for simple text or email inputs. I wonder if it's more future proof to have the autocomplete CodeMirror offers for code text areas and then implement it again with a popover for regular inputs.

Reply:
Yes, we use it in all form fields, subject, redirect etc. 
The "regular" single-line fields are limited in the number of characters that you can type and also the way it works currently with CodeMirror is not optimal as you can move the text to the next line and in this case the previous line won't be visible. We need to think about the pattern here, the potential solution might be to have "auto" height on the fields, but then you can't build all types of form UI with this approach as you can easily break the UI.